File: AD-HOC/2423/main.dart
import 'dart:io';
import 'dart:math';

int maxCakes(int a, int b, int c) {
  return min(min(a ~/ 2, b ~/ 3), c ~/ 5);
}

void main() {
  var [a, b, c] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  print(maxCakes(a, b, c));
}

File: AD-HOC/2423/main.js
function maxCakes(a, b, c) {
    return Math.min(parseInt(a / 2), parseInt(b / 3), parseInt(c / 5));
}

let [a, b, c] = require('fs').readFileSync('/dev/stdin', 'utf-8').split(' ').map(Number);
console.log(maxCakes(a, b, c));
